problem installing 3.5.0 by npx

Hi,
after run the command on my server
ubuntu : 20.04
node: v14.15.5
npm: 7.5.4

`npx create-strapi-app my-project --quickstart
`
I get

```
Creating a new Strapi application at /root/my-project.

Creating a quickstart project.
Creating files.
Error while installing dependencies:
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! While resolving: my-project@0.1.0
npm ERR! Found: knex@0.19.5
npm ERR! node_modules/knex
npm ERR!   knex@"<0.20.0" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer knex@"^0.20.0" from strapi-connector-bookshelf@3.5.0
npm ERR! node_modules/strapi-connector-bookshelf
npm ERR!   strapi-connector-bookshelf@"3.5.0" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR!
npm ERR! See /root/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2021-02-18T17_27_16_804Z-debug.log
npm ERR! See /root/.npm/eresolve-report.txt for a full report.

 Keep trying!

Oh, it seems that you encountered errors while installing dependencies in your p                                                       roject.
Don't give up, your project was created correctly.
Fix the issues mentioned in the installation errors and try to run the following                                                        command:

cd /root/my-project && npm install

npm ERR! code 1
npm ERR! path /root
npm ERR! command failed
npm ERR! command sh -c create-strapi-app "my-project" "--quickstart"

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2021-02-18T17_27_17_934Z-debug.log
```

```
npm ERR! Could not resolve dependency:
npm ERR! peer knex@"^0.20.0" from strapi-connector-bookshelf@3.5.0
```

Can you check your npm registry settings to see if you are using a 3rd party registry or proxy?

cat /root/.npm/_logs/2021-02-19T13_29_19_640Z-debug.log
`49 verbose stack Error: unable to resolve dependency tree
49 verbose stack     at Arborist.[failPeerConflict] (/usr/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/build-ideal-tree.js:1136:25)
49 verbose stack     at Arborist.[loadPeerSet] (/usr/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/build-ideal-tree.js:1116:34)
49 verbose stack     at async Arborist.[buildDepStep] (/usr/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/build-ideal-tree.js:834:11)
49 verbose stack     at async Arborist.buildIdealTree (/usr/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/build-ideal-tree.js:209:7)
49 verbose stack     at async Promise.all (index 1)
49 verbose stack     at async Arborist.reify (/usr/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/reify.js:127:5)
49 verbose stack     at async install (/usr/lib/node_modules/npm/lib/install.js:38:3)
50 verbose cwd /home/my-project
51 verbose Linux 5.4.0-65-generic
52 verbose argv "/usr/bin/node" "/usr/bin/npm" "install" "--production" "--no-optional"
53 verbose node v14.15.5
54 verbose npm  v7.5.4
55 error code ERESOLVE
56 error ERESOLVE unable to resolve dependency tree
57 error
58 error While resolving: my-project@0.1.0
58 error Found: knex@0.19.5
58 error node_modules/knex
58 error   knex@"<0.20.0" from the root project
58 error
58 error Could not resolve dependency:
58 error peer knex@"^0.20.0" from strapi-connector-bookshelf@3.5.0
58 error node_modules/strapi-connector-bookshelf
58 error   strapi-connector-bookshelf@"3.5.0" from the root project
58 error
58 error Fix the upstream dependency conflict, or retry
58 error this command with --force, or --legacy-peer-deps
58 error to accept an incorrect (and potentially broken) dependency resolution.
58 error
58 error See /root/.npm/eresolve-report.txt for a full report.
59 verbose exit 1
`

+1, Having exactly same problem even using custom setup mode.
<details>
<summary>Log</summary>

```
Creating a new Strapi application at /home/igormahov/dev/strapi/test-app.

? Choose your installation type Custom (manual settings)
? Choose your default database client postgres
? Database name: strapi
? Host: localhost
? Port: 5432
? Username: postgres
? Password: *
? Enable SSL connection: No

⛔️ Connection test failed: Command failed: npm install --prefix /tmp/strapi60ac1159482a strapi-connector-bookshelf@3.5.0 knex@<0.20.0 pg@latest
npm notice 
npm notice New patch version of npm available! 7.5.2 -> 7.5.4
npm notice Changelog: <https://github.com/npm/cli/releases/tag/v7.5.4>
npm notice Run `npm install -g npm@7.5.4` to update!
npm notice 
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: undefined@undefined
npm ERR! Found: knex@0.19.5
npm ERR! node_modules/knex
npm ERR!   knex@"<0.20.0" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer knex@"^0.20.0" from strapi-connector-bookshelf@3.5.0
npm ERR! node_modules/strapi-connector-bookshelf
npm ERR!   strapi-connector-bookshelf@"3.5.0" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /home/igormahov/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/igormahov/.npm/_logs/2021-02-19T16_49_19_420Z-debug.log



Retrying...
```
</details>

OS: Debian Unstable (bullseye)
Node version: v12.20.1

+1, I'm having the same problem.

* node: v12.18.2
* npm: 7.5.1
* os: mac os 10.15.7

```
? Choose your installation type Custom (manual settings)
? Choose your default database client postgres
? Database name: try-strapi-350
? Host: 127.0.0.1
? Port: 5432
? Username:
? Password:
? Enable SSL connection: No

⛔️ Connection test failed: Command failed: npm install --prefix /var/folders/dt/wq6c3gpj2cl2s8vdfk_23p15qbkwsh/T/strapi874de1e4501c strapi-connector-bookshelf@3.5.0 knex@<0.20.0 pg@latest
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! While resolving: undefined@undefined
npm ERR! Found: knex@0.19.5
npm ERR! node_modules/knex
npm ERR!   knex@"<0.20.0" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer knex@"^0.20.0" from strapi-connector-bookshelf@3.5.0
npm ERR! node_modules/strapi-connector-bookshelf
npm ERR!   strapi-connector-bookshelf@"3.5.0" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR!
```

might the above error be related to the below?

https://github.com/strapi/strapi/blob/6d252b12e103fecf5771424e2379a5a1c8642866/packages/strapi-generate-new/lib/utils/db-client-dependencies.js#L20-L26

strapi-connector-bookshelf@3.5.0 requires knex "^0.20.0",
but "custom-setup" might try to test db connection with the above dependencies ( knex "<0.20.0" )
( and might confilicts )

I appreciate it might be helpful to solve this problem.
thanks in advance.

--- a/examples/getstarted/package.json
+++ b/examples/getstarted/package.json
@@ -12,11 +12,11 @@
     "strapi": "strapi"
   },
   "dependencies": {
-    "knex": "^0.21.14",
+    "knex": "0.21.18",
     "lodash": "4.17.19",
-    "mysql": "^2.17.1",
+    "mysql": "2.18.1",
     "pg": "8.5.1",
-    "sqlite3": "^5.0.1",
+    "sqlite3": "5.0.0",
     "strapi": "3.5.1",
     "strapi-admin": "3.5.1",
     "strapi-connector-bookshelf": "3.5.1",

--- a/packages/strapi-connector-bookshelf/package.json
+++ b/packages/strapi-connector-bookshelf/package.json
@@ -26,7 +26,7 @@
     "strapi-utils": "3.5.1"
   },
   "peerDependencies": {
-    "knex": "^0.20.0"
+    "knex": "0.21.18"
   },
   "strapi": {
     "dependencies": [

--- a/packages/strapi-generate-new/lib/utils/db-client-dependencies.js
+++ b/packages/strapi-generate-new/lib/utils/db-client-dependencies.js
@@ -1,9 +1,9 @@
 'use strict';
 
 const sqlClientModule = {
-  sqlite: 'sqlite3',
-  postgres: 'pg',
-  mysql: 'mysql',
+  sqlite: { sqlite3: '5.0.0' },
+  postgres: { pg: '8.5.1' },
+  mysql: { mysql: '2.18.1' },
 };
 
 /**
@@ -12,17 +12,12 @@ const sqlClientModule = {
 module.exports = ({ scope, client }) => {
   switch (client) {
     case 'sqlite':
-      return {
-        'strapi-connector-bookshelf': scope.strapiVersion,
-        knex: '<0.20.0',
-        [sqlClientModule[client]]: '5.0.0',
-      };
     case 'postgres':
     case 'mysql':
       return {
         'strapi-connector-bookshelf': scope.strapiVersion,
-        knex: '<0.20.0',
-        [sqlClientModule[client]]: 'latest',
+        knex: '0.21.18',
+        ...sqlClientModule[client],
       };
     case 'mongo':
       return {

